query/stdlib/influxdata/influxdb/storage.go

```diff
@@ -148,18 +148,15 @@ type TableIterator interface {
 type ReadWindowAggregateSpec struct {
 	ReadFilterSpec
 	WindowEvery int64
-	Aggregates  []plan.ProcedureKind
+	Aggregates  []string
 }
 
-// WindowAggregateCapability describes what is supported by WindowAggregateReader.
-type WindowAggregateCapability struct{}
-
 // WindowAggregateReader implements the WindowAggregate capability.
 type WindowAggregateReader interface {
 	// HasWindowAggregateCapability will test if this Reader source supports the ReadWindowAggregate capability.
 	// If WindowAggregateCapability is passed to the method, then the struct
 	// is filled with a detailed list of what the RPC call supports.
-	HasWindowAggregateCapability(ctx context.Context, capability ...*WindowAggregateCapability) bool
+	HasWindowAggregateCapability(ctx context.Context, capability ...*reads.WindowAggregateCapability) bool
```

Comment:
This shouldn't be here. We need to duplicate this struct so we don't have an explicit dependency on the reads package. Please keep it as it was.

Reply:
Can you explain why we cannot have this dependency?

My understanding is that this `WindowAggregateCapability` struct in the future will contain flags like:
```
type WindowAggregateCapability struct {
    SupportA bool
    SupportB bool
    ....
}
```

If we keep two separate `WindowAggregateCapability` struct definitions, conversion code is needed in this function implementation to convert from `reads.WindowAggregateCapability` to `influxdb.WindowAggregateCapability`.  Dependency on the `reads` package is inevitable, but the implementation of this function is in a different package though.

Reply:
My thought process on this is that we shouldn't have this dependency as part of the interface because it assumes the reader is the specific reader from `reads.Store` and there's no other part of the interface that is dependent on that package. I can make an `influxdb.Reader` and have it live completely within the flux code. The implementation is in a separate package.

In general, I think it's an anti-pattern if an interface has a dependency on structs from other packages.

We can potentially bring this refactor to a different PR. I'd like to discuss it more, but I don't want to block work that is dependent on this PR.